codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/ApiGatewayExportsNullabilityExceptionIntegration.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.BooleanNode;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NumberNode;
+import software.amazon.smithy.model.node.StringNode;
+import software.amazon.smithy.model.shapes.BooleanShape;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.NumberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.DefaultTrait;
+import software.amazon.smithy.model.transform.ModelTransformer;
+import software.amazon.smithy.utils.IoUtils;
+import software.amazon.smithy.utils.SetUtils;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Due to internal model fixes for API Gateway (APIGW) exports, certain shapes
+ * and members that target those shapes that used to have defaults became
+ * nullable. In order to not break the existing Go v2 client API nullability,
+ * this customization adds back default values for affected shapes:
+ * 
+ * - Root Boolean shapes: false
+ * - Root Number shapes: 0
+ * - Members that target affected Root shapes: inherit defaults
+ * 
+ * The class of services affected are APIGW services and APIGW services that
+ * migrated to Smithy, seen in `APIGW_NULLABILITY_EXCEPTION_SERVICES`.
+ * 
+ * A "snapshot" of root-level shapes are captured in
+ * `APIGW_exports_nullability_exceptions.json`, a mapping of service shape IDs
+ * to a list of affected root-level shapes.
+ */
+@SmithyInternalApi
+public class ApiGatewayExportsNullabilityExceptionIntegration implements GoIntegration {
+    private static final Logger LOGGER = Logger
+            .getLogger(ApiGatewayExportsNullabilityExceptionIntegration.class.getName());
+    private static final String NULLABILITY_EXCEPTIONS_FILE = "APIGW_exports_nullability_exceptions.json";
+    private static final Set<ShapeId> APIGW_NULLABILITY_EXCEPTION_SERVICES = SetUtils.of(
```

Comment:
Would the current approach load and parse this file multiple times when building the Go SDK? Or is it loaded just once?

Reply:
Loaded per affected service model, so 22 times.

I also thought about splitting the file so that each service has its own file, but the performance benefit is marginal.